tvtk/tests/test_visual.py

```diff
@@ -0,0 +1,269 @@
+import unittest
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+from tvtk.tools import visual
+
+def get_bounds(pos, sz):
+    pos = np.asarray(pos)
+    sz = np.asarray(sz)
+    diff = sz*0.5
+    bounds = np.zeros(6, dtype=float)
+    bounds[::2] = pos - diff
+    bounds[1::2] = pos + diff
+    return bounds
+
+class DummyScene:
+    def __init__(self):
+        self.actors = []
+
+    def add_actors(self, *args):
+        self.actors.extend(args)
+
+    def remove_actors(self, *args):
+        for arg in args:
+            self.actors.remove(arg)
+
+    def render(self):
+        pass
+
+    def reset_zoom(self):
+        pass
+
+
+class DummyViewer:
+    def __init__(self):
+        self.scene = DummyScene()
+
+
+class TestVisual(unittest.TestCase):
+    def setUp(self):
+        self.viewer = DummyViewer()
+        visual.set_viewer(self.viewer)
+
+    def tearDown(self):
+        visual.set_viewer(None)
+
+    def test_ring(self):
+        # Given
+        r = visual.ring()
+
+        # Then
+        assert_allclose(r.polydata.bounds, (0.0, 0.0, -0.5, 0.5, -0.5, 0.5))
+        self.assertEqual(r.radius, 0.5)
+        self.assertEqual(r.tube.radius, 0.01)
+
+        # Given
+        r = visual.ring(radius=1.0, pos=(1.0, 1.0, 1.0))
+
+        # When/Then
+        self.assertEqual(r.radius, 1.0)
+        self.assertEqual(r.thickness, 0.01)
+        self.assertEqual(r.tube.radius, 0.01)
+        self.assertTrue(np.allclose(r.pos, (1.0, 1.0, 1.0)))
```

Comment:
That one slipped, I have used it right below...  Is it preferred to use `assert_array_almost_equal` or `assert_allclose`??